framework/docker/docker_options.go

```diff
@@ -72,3 +72,94 @@ func WithPerLightNodeConfig(nodeConfigs map[int]*DANodeConfig) ConfigOption {
 		cfg.DataAvailabilityNetworkConfig.LightNodeConfigs = nodeConfigs
 	}
 }
+
+// SIMPLE: One-liner to configure all DA node ports
+func WithDANodePorts(rpcPort, p2pPort string) ConfigOption {
+	return func(cfg *Config) {
+		if cfg.DataAvailabilityNetworkConfig == nil {
+			cfg.DataAvailabilityNetworkConfig = &DataAvailabilityNetworkConfig{}
+		}
+		cfg.DataAvailabilityNetworkConfig.DefaultRPCPort = rpcPort
+		cfg.DataAvailabilityNetworkConfig.DefaultP2PPort = p2pPort
+	}
+}
+
+// SIMPLE: One-liner to configure core connection ports
+func WithDANodeCoreConnection(rpcPort, grpcPort string) ConfigOption {
+	return func(cfg *Config) {
+		if cfg.DataAvailabilityNetworkConfig == nil {
+			cfg.DataAvailabilityNetworkConfig = &DataAvailabilityNetworkConfig{}
+		}
+		cfg.DataAvailabilityNetworkConfig.DefaultCoreRPCPort = rpcPort
+		cfg.DataAvailabilityNetworkConfig.DefaultCoreGRPCPort = grpcPort
+	}
+}
+
+// SIMPLE: Configure everything in one call
+func WithCustomPortsSetup(daRPC, daP2P, coreRPC, coreGRPC string) ConfigOption {
+	return func(cfg *Config) {
+		if cfg.DataAvailabilityNetworkConfig == nil {
+			cfg.DataAvailabilityNetworkConfig = &DataAvailabilityNetworkConfig{}
+		}
+		cfg.DataAvailabilityNetworkConfig.DefaultRPCPort = daRPC
+		cfg.DataAvailabilityNetworkConfig.DefaultP2PPort = daP2P
+		cfg.DataAvailabilityNetworkConfig.DefaultCoreRPCPort = coreRPC
+		cfg.DataAvailabilityNetworkConfig.DefaultCoreGRPCPort = coreGRPC
+	}
+}
+
+// SIMPLE: Per-node-type configuration for bridge nodes
+func WithBridgeNodePorts(nodeIndex int, rpcPort, p2pPort string) ConfigOption {
```

Comment:
rather than having all these `With<NodeType>NodePorts` functions, it might be cleaner to accept a node type as an argument WDYT?

```go
WithNodePorts(types.BridgeNode, /* remaining args */)
```